From e41e259eacffbf492d832a96cb200cb8f5cfc4e3 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Sat, 22 Jul 2017 15:31:13 +0800
Subject: [PATCH] sound: simple-card: only set det_invert when det is valid

If the kernel option CONFIG_GPIOLIB is enabled, everything is well.
But if this option is disabled, we shouldn't set gpio_*_det_invert.
So add a judgement gpio_is_valid() before set these variables.
Else we would get the below compile warning:

sound/soc/generic/simple-card.c:406:21: warning:
 'flags' may be used uninitialized in this function
 [-Wmaybe-uninitialized] enum of_gpio_flags flags;
 ^

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 sound/soc/generic/simple-card.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/generic/simple-card.c b/sound/soc/generic/simple-card.c
index 33feee9..2db540e 100644
--- a/sound/soc/generic/simple-card.c
+++ b/sound/soc/generic/simple-card.c
@@ -461,15 +461,17 @@ static int asoc_simple_card_parse_of(struct device_node *node,
 
 	priv->gpio_hp_det = of_get_named_gpio_flags(node,
 				"simple-audio-card,hp-det-gpio", 0, &flags);
-	priv->gpio_hp_det_invert = !!(flags & OF_GPIO_ACTIVE_LOW);
 	if (priv->gpio_hp_det == -EPROBE_DEFER)
 		return -EPROBE_DEFER;
+	if (gpio_is_valid(priv->gpio_hp_det))
+		priv->gpio_hp_det_invert = !!(flags & OF_GPIO_ACTIVE_LOW);
 
 	priv->gpio_mic_det = of_get_named_gpio_flags(node,
 				"simple-audio-card,mic-det-gpio", 0, &flags);
-	priv->gpio_mic_det_invert = !!(flags & OF_GPIO_ACTIVE_LOW);
 	if (priv->gpio_mic_det == -EPROBE_DEFER)
 		return -EPROBE_DEFER;
+	if (gpio_is_valid(priv->gpio_mic_det))
+		priv->gpio_mic_det_invert = !!(flags & OF_GPIO_ACTIVE_LOW);
 
 	if (!priv->snd_card.name)
 		priv->snd_card.name = priv->snd_card.dai_link->name;
-- 
1.7.5.4

